Library/Homebrew/download_strategy.rb

```diff
@@ -195,14 +211,15 @@ def initialize(url, name, version, **meta)
   # Download and cache the repository at {#cached_location}.
   #
   # @api public
+  sig { params(timeout: T.nilable(T.any(Integer, Float))).void }
   def fetch(timeout: nil)
     end_time = Time.now + timeout if timeout
 
     ohai "Cloning #{url}"
 
     if cached_location.exist? && repo_valid?
       puts "Updating #{cached_location}"
-      update(timeout: timeout)
+      update(timeout: end_time)
```

Comment:
@MikeMcQuaid This is a minor bug that I fixed while working on annotations.

Reply:
@Kentzo Please split this into a separate PR and keep this PR focused entirely on the `sudo` changes, thanks.